webtau-http/src/test/java/com/twosigma/webtau/http/HttpJavaTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2018 TWO SIGMA OPEN SOURCE, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.twosigma.webtau.http;
+
+import com.twosigma.webtau.Ddjt;
+import com.twosigma.webtau.http.config.HttpConfiguration;
+import com.twosigma.webtau.http.config.HttpConfigurations;
+import com.twosigma.webtau.utils.CollectionUtils;
+import com.twosigma.webtau.utils.UrlUtils;
+import org.junit.*;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+import static com.twosigma.webtau.Ddjt.*;
+import static com.twosigma.webtau.http.Http.http;
+import static com.twosigma.webtau.utils.CollectionUtils.createMap;
+
+public class HttpJavaTest implements HttpConfiguration  {
+    private static final HttpTestDataServer testServer = new HttpTestDataServer();
+
+    private static final int PORT = 7824;
+
+    @BeforeClass
+    public static void startServer() {
+        testServer.start(PORT);
```

Comment:
At some point we probably should change this to start on a random port and return the port number otherwise we'll inevitably get some clashes.

Reply:
yeah... Is there an easy way that is built in in a jetty that I am using?

Reply:
Pretty sure Jetty allows you to specify port 0 and then it'll pick a random port.  Its server class should have a `port()` or `httpPort()` method (can't remember which).

Reply:
I think with Jetty you may have to actually get the connectors from the server and then their port rather than asking the server itself but in any case it's definitely doable.

Reply:
https://github.com/twosigma/webtau/issues/181